static/skin/index.js

```diff
@@ -209,11 +213,17 @@
         });
     }
 
-    async function loadAndDisplayOptions(nodeQuery, query) {
-        // currently taking an object in place of query, will replace it with query while fetching data from backend later on.
-        document.querySelector(nodeQuery).innerHTML += Object.keys(query)
-            .map((option) => {return `<option value='${option}'>${htmlEncode(query[option])}</option>`})
-        .join('');
+    async function loadAndDisplayOptions(nodeQuery, query, type = '') {
+        await fetch(query).then(async (resp) => {
+            const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
+            let optionStr = '';
+            data.querySelectorAll('entry').forEach(entry => {
+                const title = getInnerHtml(entry, 'title');
+                const value = type === 'lang' ? getInnerHtml(entry, 'language') : title;
```

Comment:
Special handling of languages doesn't feel right. Why don't you make the `type` parameter of `loadAndDisplayOptions()` mandatory and pass it to `getInnerHtml()` unmodified?

Reply:
@veloman-yunkan that doesn't seem right to me.....there is no meaning of adding `type` to `getInnerHtml()` it's all because the value for langugae options need to be iso-3 value of language while for categories title and value are same.
can you be more specific please.

Reply:
I meant having

```
const value = getInnerHtml(entry, type);
```

in `loadAndDisplayOptions()` and calling the latter as follows:

```
await loadAndDisplayOptions('#languageFilter', `${root}/catalog/v2/languages`, 'language');
await loadAndDisplayOptions('#categoryFilter', `${root}/catalog/v2/categories`, 'title');
```

Then the `type` parameter of `loadAndDisplayOptions()` should be renamed too (`entry_node_name`?).